contracts/PinningManager.sol

```diff
@@ -162,17 +158,14 @@ contract PinningManager {
     @notice request to fill a storageOffer. After requesting, an offer must be accepted by provider to become active.
     @dev if Request was active before, is expired and final payout is not yet done, final payout can be triggered by proposer here.
     The to-be-pinned file's size in bytes (rounded up) must be equal in size to param size.
-    @param fileReference the reference to the to-be-pinned file (when contentManager is not used)
-    When called by a ContentManager is used, this fileReference can be a nonce (such that a contentManager smart-contract can be used to request pinning for multiple files)
+    @param fileReference the reference to the to-be-pinned file
     @param provider the provider from which is proposed to take a StorageOffer.
     @param size the size of the to-be-pinned file in bytes (rounded up).
     @param period the chosen period (seconds after which a Request can be cancelled and left-over money refunded).
-    @param contentManager if not equal to 0 address, we use this address to manage content and funds. WARNING: contentManager MUST adhere to the contentManager ABI.
     */
-    function newRequest(bytes32[] memory fileReference, address payable provider, uint120 size, uint64 period, address contentManager) public payable {
+    function newRequest(bytes32[] memory fileReference, address payable provider, uint128 size, uint64 period) public payable {
         require(period != 0, "PinningManager: period of 0 not allowed");
-        bool usesContentManager = contentManager != address(0);
-        bytes32 requestReference = getRequestReference(msg.sender, fileReference, usesContentManager);
+        bytes32 requestReference = getRequestReference(msg.sender, fileReference);
         uint64 chosenPrice = offerRegistry[provider].prices[period];
```

Comment:
Bit unrelated but I believe that the `period` should be named `periodIndex` because this implies that the actual period should be passed but here it is used to index `prices` array which won't work if actual period is passed.

Reply:
No, it's the actual period which is passed. E.g, if you want to have the price, corresponding to a period of 2 weeks, you pass the amount of seconds that fit in two weeks.
As documented:
`@param period the chosen period (seconds after which a Request can be cancelled and left-over money refunded).`

Reply:
Well then I believe you have a bug in your code. 

I believe that `.prices` is an array, right? But you are using `period` as an array's index, which does not make sense if the `period` is a number of seconds for the period as you mentioned and not the `periodIndex`.

Maybe I am missing something?

Reply:
`.price` is a mapping (see line 25):
`mapping(uint64 => uint64) prices;`
It is only an array when setting this mapping, to allow for passing in multiple period-price pairs. But they are stored as a mapping, not as an array.